### PR TITLE
38 heatmaps nonsobiose

### PR DESCRIPTION
**Pull request by nonsobiose**

### CODE REVIEW

File: backend/functions/routes/heatmaps-coordinates.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+const {app} = require('../setups/setup.js');
+const MongoClient = require('mongodb').MongoClient;
+
+const DB = '947g5';
+const COLLECTION = 'addressbase';
+const USERNAME = 'react-read';
+const PASSWORD = '31vTs8wfBB9KvfIy';
```
Comment (georgeracu) [inline backend/functions/routes/heatmaps-coordinates.js]: Why do we have credentials in plain sight? We should have them encrypted.

File: geolocation/geolocation.js [hunk 2]
```diff
@@ -52,7 +54,97 @@ async function getCurrentGeolocation(handleSuccess, handleFailure) {
...
 
+/**
+ * This function retrieves coordinates to used in displaying in HeatMaps
+ * @param endpoint
+ * @param coords
+ * @param handleState
+ * @returns {Promise<void>}
+ */
+async function getHeatMapsCoordinates(endpoint, coords, handleState) {
+  console.log('2');
```
Comment (georgeracu) [inline geolocation/geolocation.js]: We do not do `console.log` in production code.

File: ios/Podfile.lock [hunk 3]
```diff
@@ -325,6 +332,8 @@ PODS:
     - React-cxxreact (= 0.61.2)
     - React-jsi (= 0.61.2)
     - ReactCommon/jscallinvoker (= 0.61.2)
+  - ReactNativeAndroidLocationSwitch (1.0.0):
```
Comment (georgeracu) [inline ios/Podfile.lock]: Is this pod Android only?

File: ios/Podfile.lock [hunk 4]
```diff
@@ -371,6 +380,7 @@ DEPENDENCIES:
   - React-RCTVibration (from `../node_modules/react-native/Libraries/Vibration`)
   - ReactCommon/jscallinvoker (from `../node_modules/react-native/ReactCommon`)
   - ReactCommon/turbomodule/core (from `../node_modules/react-native/ReactCommon`)
+  - ReactNativeAndroidLocationSwitch (from `../node_modules/react-native-location-switch/ios`)
```
Comment (georgeracu) [inline ios/Podfile.lock]: Same here regarding the Android only switch.

File: utils/constants.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const LOGS_ENDPOINT =
+  'https://us-central1-test-947g5.cloudfunctions.net/logs/log';
```
Comment (georgeracu) [inline utils/constants.js]: Needs to be encrypted.

File: utils/constants.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+const HEATMAPS_ENDPOINT =
+  'https://us-central1-test-947g5.cloudfunctions.net/heatMapsCoordinates/heatmaps';
```
Comment (georgeracu) [inline utils/constants.js]: Needs to be encrypted.

File: App.js [hunk 7]
```diff
@@ -1,8 +1,16 @@
+/**
+ *
+ * @format
+ * @flow
+ */
+
```
Comment (georgeracu) [inline App.js]: Not necessary comments.

File: App.js [hunk 8]
```diff
@@ -13,24 +21,34 @@ export default class App extends Component {
...
-    this.LOGS_ENDPOINT =
-      'https://us-central1-test-947g5.cloudfunctions.net/logs/log';
   }
 
   /**
    * This method updates the state object with the response from the Geolocation API
    * @param geoCoords
    */
   onGeolocationSuccess = async geoCoords => {
+    console.log('1');
```
Comment (georgeracu) [inline App.js]: Why do we need console.log?

File: backend/functions/routes/heatmaps-coordinates.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+app.post('/heatmaps', (req, res) => {
+  const uri = `mongodb+srv://${USERNAME}:${PASSWORD}@cluster0-cbuck.mongodb.net/test?retryWrites=true&w=majority`;
```
Comment (georgeracu) [inline backend/functions/routes/heatmaps-coordinates.js]: We do not have to have this URL here if we plan to use Firebase functions to call MongoDB. Also the URL needs to be encrypted.

File: utils/constants.js (hunk 5 above)
Comment (nonsobiose) [reply utils/constants.js]: Using a cryptography library or environment variables?

File: backend/functions/routes/heatmaps-coordinates.js (hunk 9 above)
Comment (nonsobiose) [reply backend/functions/routes/heatmaps-coordinates.js]: This is a firebase function.

**Comment by nonsobiose** [comment]
Was cloned by @georgeracu into another branch to fix a CI/CD issue, then merged into Master.

**Pull request closed without merging**